code/modules/virus2/dishincubator.dm

```diff
@@ -173,21 +170,21 @@
 	onclose(user, "dish_incubator")
 
 /obj/machinery/disease2/incubator/process()
-	if(dish && on && dish.virus2)
+	if(on)
 		use_power(50,EQUIP)
 		if(!powered(EQUIP))
 			on = 0
 			icon_state = "incubator"
-		if(foodsupply)
+		if(dish && dish.growth >= INCUBATOR_MAX_SIZE)
+			if(icon_state != "incubator_fed")
+				icon_state = "incubator_fed"
+			if(last_notice + FED_PING_DELAY < world.time)
+				last_notice = world.time
+				alert_noise("ping")
+		if(dish && dish.virus2 && foodsupply)
```

Comment:
Indent all of these conditionals and stick them all under a `if(dish && dish.virus2)`. No sense in duplicating them all.